src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -575,12 +601,13 @@ ShardLength(uint64 shardId)
 
 
 /*
- * NodeHasActiveShardPlacements returns whether any active shards are placed on this node
+ * NodeHasShardPlacements returns whether any active shards are placed on the group
+ * this node is a part of.
  */
 bool
-NodeHasActiveShardPlacements(char *nodeName, int32 nodePort)
+NodeHasShardPlacements(char *nodeName, int32 nodePort, bool onlyConsiderActivePlacements)
 {
-	const int scanKeyCount = 3;
+	const int scanKeyCount = (onlyConsiderActivePlacements ? 2 : 1);
```

Comment:
I'd move this down to the if block considering onlyConsiderActivePlacements - unnecessary to have to look back up for it...